packages/vite/src/node/plugins/define.ts

```diff
@@ -62,7 +63,12 @@ export function definePlugin(config: ResolvedConfig): Plugin {
       ...(isNeedProcessEnv ? processNodeEnv : {}),
       ...userDefine,
       ...importMetaKeys,
-      ...processEnv
+      ...processEnv,
+      ...(isWorker
+        ? {
+            'import.meta.url': 'self.location.href'
+          }
+        : {})
```

Comment:
I think it would be a bit cleaner if we move this code to https://github.com/vitejs/vite/blob/b521aa4b8b1544b1f0806736d478d4cfdff6e750/packages/vite/src/node/plugins/define.ts#L41-L45

so the `import.meta` stuff are at one spot.